UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common.UnitTests/HealthCheck/FileShareServiceHealthCheckTest.cs

```diff
@@ -0,0 +1,81 @@
+﻿using FakeItEasy;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.HealthCheck;
+using UKHO.ExchangeSetService.Common.Helpers;
+
+namespace UKHO.ExchangeSetService.Common.UnitTests.HealthCheck
+{
+    public class FileShareServiceHealthCheckTest
+    {
+        private ILogger<FileShareService> fakeLogger;
+        private IOptions<FileShareServiceConfiguration> fakeFileShareConfig;
+        private IAuthTokenProvider fakeAuthTokenProvider;
+        private IFileShareServiceClient fakeFileShareServiceClient;
+        private FileShareServiceHealthCheck fileShareServiceHealthCheck;
+
+        [SetUp]
+        public void Setup()
+        {
+            this.fakeLogger = A.Fake<ILogger<FileShareService>>();
+            this.fakeAuthTokenProvider = A.Fake<IAuthTokenProvider>();
+            this.fakeFileShareConfig = Options.Create(new FileShareServiceConfiguration()
+            { BaseUrl = "http://tempuri.org", CellName = "DE260001", EditionNumber = "1", Limit = 1, Start = 0, ProductCode = "AVCS", ProductLimit = 4, UpdateNumber = "0", UpdateNumberLimit = 10 });
+            this.fakeFileShareServiceClient = A.Fake<IFileShareServiceClient>();
+
+            fileShareServiceHealthCheck = new FileShareServiceHealthCheck(fakeFileShareServiceClient, fakeAuthTokenProvider, fakeFileShareConfig, fakeLogger);
+        }
+
+        #region GetFakeToken
+        private static string GetFakeToken()
+        {
+            return "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ0ZXN0IHNlcnZlciIsImlhdCI6MTU1ODMyOTg2MCwiZXhwIjoxNTg5OTUyMjYwLCJhdWQiOiJ3d3cudGVzdC5jb20iLCJzdWIiOiJ0ZXN0dXNlckB0ZXN0LmNvbSIsIm9pZCI6IjE0Y2I3N2RjLTFiYTUtNDcxZC1hY2Y1LWEwNDBkMTM4YmFhOSJ9.uOPTbf2Tg6M2OIC6bPHsBAOUuFIuCIzQL_MV3qV6agc";
+        }
+        #endregion
+
+        [Test]
+        public async Task WhenFSSClientReturnsOtherThan200And304_ThenFileShareServiceIsUnhealthy()
```

Comment:
Remove 304 from test case name as FSS client doesn't return 304 status code for search query.

Reply:
Fixed